docs/distro-compat.md

```diff
@@ -3,41 +3,125 @@
 ## Contents
 
 -   [Supported Linux Distros](#supported-linux-distros)
--   [Ubuntu Support](#ubuntu-support)
+-   [Sysbox Operational Modes](#Sysbox-Operational-Modes)
+-   [Kernel Upgrade Procedures](#Kernel-Upgrade-Procedures)
 
 ## Supported Linux Distros
 
-Sysbox relies on functionality that is currently only present in Ubuntu Linux.
-As a result Ubuntu is the only distro supported at this time. We are working
-on adding support for more distros.
+Sysbox relies on functionality that is only present in relatively modern Linux
+kernels.
 
-## Ubuntu Support
+In Ubuntu's case, these requirements are met starting with Kernel 5.0. For
+the rest of the Sysbox's supported distributions, Kernel 5.5+ is required.
 
-Sysbox requires recent Ubuntu versions:
+The following table summarizes the Linux distributions being supported as well
+as the operational modes (`shiftfs` vs `userns-remap`) being allowed. For
+scenarios where a kernel upgrade is required, refer to the corresponding
+upgrade procedure further below.
 
--   Ubuntu 20.04 "Focal Fossa"
--   Ubuntu 19.10 "Eoan Ermine"
--   Ubuntu 18.04 "Bionic Beaver" (point releases older than 18.04.4 require kernel upgrade)
+<p align="center">
+    <img alt="sysbox" src="figures/distro-support-matrix.png" width="1000x" />
+</p>
 
-These versions carry some new Linux kernel features that Sysbox relies on to
-create the system containers.
+* Note 1: Ubuntu Bionic requires no kernel upgrade in scenarios where relatively
+recent releases have been installed (i.e. 18.04.4+). If that's not the
+case, user will need to upgrade the kernel. Refer to [Ubuntu's kernel-upgrade](#Ubuntu-Kernel-Upgrade)
+procedure for details.
 
-**NOTE:** If you have a relatively old Ubuntu 18.04 release (e.g. 18.04.3), you need to upgrade the kernel to >= 5.0.
-We recommend using Ubuntu's [LTS-enablement](https://wiki.ubuntu.com/Kernel/LTSEnablementStack)
-package to do the upgrade as follows:
+* Note 2: Fedora 31 requires no kernel upgrade assuming that user is running a recent
+installation (i.e. kernel 5.5+ is deployed). Otherwise, a kernel-upgrade will be
+expected. Refer to [Fedora's kernel-upgrade](#Fedora-kernel-upgrade) procedure for details.
+
+## Sysbox Operational Modes
```

Comment:
This section is basically repeating what's in the user-guide. Let's point users to the user-guide.

Reply:
Mmm, don't agree here, this section was already present in this doc, i'm simply extending it to complement the technical details described in the user-guide. The perspective here is different than the user-guide's one; the goal is for the user to quickly understand the installation requirements. See my latest patch.

Reply:
Let's talk later. I think we should remove the operational modes from here (leave that in the user-guide as it is), and replace the columns for shiftfs and userns-remap in the distro support matrix with a single column that say "requires Docker userns-remap mode".  That way people we tell users directly what's needed  (rather than going into details of shiftfs).

Reply:
Already discussed.